tracer/src/Datadog.Trace/Agent/TraceSamplers/RareSampler.cs

```diff
@@ -0,0 +1,71 @@
+// <copyright file="RareSampler.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Agent.TraceSamplers
+{
+    internal class RareSampler : ITraceSampler
+    {
+        private const string RareKey = "_dd.rare";
+        private readonly HashSet<StatsAggregationKey> _keys = new();
+
+        public bool Sample(ArraySegment<Span> trace)
+        {
+            bool priorityGreaterThan0 = trace.Array[trace.Offset].Context.TraceContext.SamplingPriority is int p && p > 0;
```

Comment:
Can `TraceContext` be null (`SpanContext` created from distributed context)?

Reply:
Good catch, yeah that's possible. I'll do a null-check on `TraceContext`

Reply:
Fixed in 45c222742498c45a834882cb5c2cd09003b2126b